dandiapi/api/views/asset.py

```diff
@@ -94,19 +99,15 @@ def raise_if_unauthorized(self):
         # Clients must be authenticated to access it
         if not self.request.user.is_authenticated:
             raise NotAuthenticated
+        self.request.user = typing.cast(User, self.request.user)
 
         # Admins are allowed to access any embargoed asset blob
         if self.request.user.is_superuser:
             return
 
         # User must be an owner on any of the dandisets this asset belongs to
```

Comment:
The logic changed so that we're not checking the user's dandisets here to determine asset ownership, so this comment is outdated now
```suggestion
```

Reply:
How has the logic changed? `is_owned_asset` is called right below this line, which is what this comment is referring to.

Reply:
"Logic changed" was probably the wrong wording, I meant that we're not explicitly checking dandiset ownership in the code under this comment anymore, but instead just calling the opaque `is_owned_asset` function.

Reply:
I still think the comment is valid, as it describes both why we're calling `is_owned_asset`, as well as the fact that this check returns true if the user is an owner of _any_ dandiset this asset belongs to.

Reply:
Fair enough, I don't have any objections to keeping it